src/javis_viewer.jl

```diff
@@ -0,0 +1,230 @@
+"""
+    _draw_image(video::Video, actions::Vector, frame::Int, canvas::Gtk.Canvas,
+    img_dims::Vector)
+
+Internal function to create an image that is drawn on a Gtk Canvas.
+"""
+function _draw_image(
+    video::Video,
+    actions::Vector,
+    frame::Int,
+    canvas::Gtk.Canvas,
+    img_dims::Vector,
+)
+    @guarded draw(canvas) do widget
+        # Gets a specific frame from graphic; transposed due to returned matrix
+        frame_mat = transpose(get_javis_frame(video, actions, frame))
+
+        # Gets the correct Canvas context to draw on
+        context = getgc(canvas)
+
+        # Uses Cairo to draw on Gtk canvas context
+        image(context, CairoImageSurface(frame_mat), 0, 0, img_dims[1], img_dims[2])
+    end
+end
+
+"""
+    _increment(video::Video, widgets::Vector, actions::Vector, dims::Vector, 
+        canvas::Gtk.Canvas, frames::Int)
+
+Increments a given value and returns the associated frame.
+"""
+function _increment(
+    video::Video,
+    widgets::Vector,
+    actions::Vector,
+    dims::Vector,
+    canvas::Gtk.Canvas,
+    frames::Int,
+)
+    # Get current frame from textbox as an Int value
+    curr_frame = parse(Int, get_gtk_property(widgets[2], :text, String))
+    if frames > curr_frame
+        push!(widgets[1], curr_frame + 1)
```

Comment:
Can you explain what widgets are so what is `widget[1]` and what is `widget[2]`?

Reply:
Added comments to explain what these two widgets represent